### PR TITLE
[JENKINS-48932] - Update plugin to make it runable with PCT

### PR DESCRIPTION
**Pull request by rsandell**
Replaces #81 with test, findbugs and checkstyle fixes.

Bumped all the way up to Jenkins Core 2.7 since even that is fairly old by now and has more findbugs annotations to adhere to.

@reviewbybees

### CODE REVIEW

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCause.java [hunk 1]
```diff
@@ -659,7 +659,7 @@ public void setLastFailedBuildUrl() {
 
         @Override
         public String getDisplayName() {
-            return null;
+            return "";
```
Comment (oleg-nenashev) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCause.java]: IIRC it can be already removed in this version... or not?

Comment (jglick) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCause.java]: Just delete this override altogether.

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java [hunk 2]
```diff
@@ -128,12 +129,19 @@ public void run() {
             if (build instanceof AbstractBuild) {
                 AbstractBuild abstractBuild = (AbstractBuild)build;
                 Node node = abstractBuild.getBuiltOn();
-                nodeName = node.getNodeName();
+                if (node != null) {
+                    nodeName = node.getNodeName();
```
Comment (oleg-nenashev) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java]: Probably you just want to fail the run, but OK

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java [hunk 3]
```diff
@@ -128,12 +129,19 @@ public void run() {
...
+                }
             }
             int timeZoneOffset = TimeZone.getDefault().getRawOffset();
             String master;
 
-            String result = build.getResult().toString();
+
+            String result = "Unknown";
+            final Result buildResult = build.getResult();
+            if (buildResult != null) {
```
Comment (oleg-nenashev) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java]: Maybe result should be "Running" according to the getResult() docs

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java [hunk 4]
```diff
@@ -310,8 +310,9 @@ public static PluginImpl getInstance() {
         PluginImpl plugin = jenkins.getPlugin(PluginImpl.class);
         if (plugin == null) {
             throw new AssertionError("Not here yet.");
+        } else {
+            return plugin;
         }
-        return plugin;
```
Comment (oleg-nenashev) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/PluginImpl.java]: hmm... FindBugs should be fine with that

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCause.java (hunk 1 above)
Comment (rsandell) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/model/FailureCause.java]: Even on 1.625 it was `@Nonnull`

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java (hunk 3 above)
Comment (rsandell) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java]: This code shouldn't execute until after the run is done (for now) so a `null` here would mean unknown and something is wrong, but I don't want to skip logging the statistics just for that.

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java (hunk 2 above)
Comment (rsandell) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java]: It is set to unknown above.

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java (hunk 3 above)
Comment (oleg-nenashev) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/statistics/StatisticsLogger.java]: Any Build Step may actually change this expectation. Better safe than sorry

**Pull request merged**